src/QirTools/qirlib/src/emit/mod.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use self::constants::Constants;
+use self::intrinsics::Intrinsics;
+use self::runtime_library::RuntimeLibrary;
+use self::types::Types;
+use inkwell::values::BasicValueEnum;
+
+use super::interop::SemanticModel;
+use std::collections::HashMap;
+use std::path::Path;
+
+pub mod constants;
+mod intrinsics;
+pub mod qir;
+mod runtime_library;
+pub mod types;
+
+pub struct Emitter {}
```

Comment:
A Rustdoc comment here as to the purpose of the `Emitter` struct may be helpful.

Reply:
Removed as it is no longer needed